File: services/meeting_agent/browser_bot/healthcheck.py
#!/usr/bin/env python3
"""
Health check script for the browser bot
"""

import os
import sys
import asyncio
import aiohttp
from datetime import datetime

async def health_check():
    """Perform health check by pinging the backend API"""
    try:
        api_base_url = os.getenv('API_BASE_URL', 'http://localhost:8000')
        api_url = f"{api_base_url}/api/v1/meetings/bot-status"
        
        async with aiohttp.ClientSession() as session:
            async with session.get(api_url, timeout=aiohttp.ClientTimeout(total=5)) as response:
                if response.status == 200:
                    print(f"[{datetime.now()}] Health check passed")
                    return True
                else:
                    print(f"[{datetime.now()}] Health check failed: {response.status}")
                    return False
                    
    except Exception as e:
        print(f"[{datetime.now()}] Health check error: {e}")
        return False

if __name__ == "__main__":
    success = asyncio.run(health_check())
    sys.exit(0 if success else 1)

File: services/meeting_agent/browser_bot/healthcheck.js
/**
 * Health check script for browser bot container
 */

const axios = require('axios');

const config = {
  apiBaseUrl: process.env.API_BASE_URL || 'http://localhost:8000',
  meetingId: process.env.MEETING_ID || 'test-meeting',
  sessionId: process.env.SESSION_ID || 'test-session'
};

async function healthCheck() {
  try {
    // Check if the bot can reach the API
    const response = await axios.get(`${config.apiBaseUrl}/health`, { timeout: 5000 });
    
    if (response.status === 200) {
      console.log('Health check passed');
      process.exit(0);
    } else {
      console.log('Health check failed: API returned non-200 status');
      process.exit(1);
    }
  } catch (error) {
    console.log('Health check failed:', error.message);
    process.exit(1);
  }
}

// Run health check
healthCheck();
